internal/envoy/route.go

```diff
@@ -292,7 +293,7 @@ func headerMatcher(headers []dag.HeaderCondition) []*envoy_api_v2_route.HeaderMa
 			header.HeaderMatchSpecifier = &envoy_api_v2_route.HeaderMatcher_ExactMatch{ExactMatch: h.Value}
 		case "contains":
 			header.HeaderMatchSpecifier = &envoy_api_v2_route.HeaderMatcher_RegexMatch{
-				RegexMatch: fmt.Sprintf(".*%s.*", h.Value),
+				RegexMatch: fmt.Sprintf(".*%s.*", regexp.QuoteMeta(h.Value)),
```

Comment:
Would it be simpler to write this as
```
RegexMatch: regex.QuoteMeta(h.value),
```
as the regex does not have `^` or `$` anchors so the contents of `h.Value` will match any position without the `.*` globs.

Reply:
But for Envoy to match properly you have to match the entire string. Without the wildcards, at beginning/end I don't think it would match.

Reply:
Would you do something for me. Could you add a helper
```
// containsMatch returns a HeaderMatchSpecifier which will match the
// supplied substring
func containsMatch(s string) *envoy_api_v2_route.HeaderMatcher_RegexMatch {
    // convert the substring s into a regular expression that matches s.
    // note that Envoy expects the expression to match the entire string, not just the substring
   // formed from s. see [the issue]
   return &envoy_api_v2_route.HeaderMatcher_RegexMatch{
 				RegexMatch: fmt.Sprintf(".*%s.*", regexp.QuoteMeta(h.Value)),
   }
}
```